abtesting/Shared/AppConfig.swift

```diff
@@ -57,6 +57,33 @@ class AppConfig: ObservableObject {
     }
   }
 
+  #if swift(>=5.5)
+    @available(iOS 15, *)
+    func updateFromRemoteConfigAsync() async {
+      let remoteConfig = RemoteConfig.remoteConfig()
+      do {
+        let status = try await remoteConfig.fetch(withExpirationDuration: 0)
+        print("Config fetch completed with status: \(status.debugDescription)")
+        do {
+          let changed = try await remoteConfig.activate()
+          let value = remoteConfig["color_scheme"].stringValue ?? "nil"
+          if changed {
+            print("Remote Config changed to: \(value)")
+            DispatchQueue.main.async {
```

Comment:
Wouldn't it be better to use @MainActor to ensure `colorScheme` is updated on the main actor?

Using DispatchQueue.main.async in combination with the new async/await features feels a bit like an anti-pattern to me.

See https://peterfriese.dev/swiftui-concurrency-essentials-part1/#updating-published-properties

Reply:
Great catch, thanks! I will include this change in a separate PR 👍

Reply:
Actually, is there an alternative other than using `@MainActor`? Since we need the app to be backward-compatible with previous versions of iOS, I am not sure how to conditionally add `@MainActor`...

Reply:
There are a couple of options:
1) You could apply it to a function that needs to run on the main actor (and the #ifdef that function)
2) Add it to the class (and #ifdef it there)
3) Call `MainAction.run` directly: https://www.avanderlee.com/swift/mainactor-dispatch-main-thread/#using-the-main-actor-directly

Reply:
Awesome, in that case I can apply it to the already conditionally-compiled async function. Thanks for the pointers!